primitives/core/src/offchain/error.rs

```diff
@@ -0,0 +1,18 @@
+#![allow(missing_docs)]
+
+use sp_std::{prelude::*, fmt, result};
+
+pub trait Error: fmt::Debug + fmt::Display {}
```

Comment:
Why not use `core::error::Error`?

Reply:
That doesn't exist afaik!

Reply:
I think we could move this trait to sp_std but I'm really not sure about this. Maybe @tomaka has some valuable insights on this

Reply:
Yeah, the `Error` trait doesn't exist in `no_std` contexts IIRC because of the unstable `Error::backtrace()` method. The replacement here looks good.

However IMO anything that contains the word "database" implies I/O, meaning in turn that it probably shouldn't be compiled to Wasm.

Reply:
In the end it was not needed at all. I removed this entire file I added.

Reply:
done